File: src/v2/api/api/Models/Entities/PaymentFailure.cs
using System.ComponentModel.DataAnnotations.Schema;


namespace Api.Models.Entities
{
    public class PaymentFailure
    {
        public long ID { get; init; }
        public long EventID { get; init; }
        public long? PersonID { get; init; }
        public long Amount { get; init; }
        public string Code { get; init; }
        public string? Decline_Code { get; init; }
        public string Message { get; init; }

    }
}

File: src/combined/ClientApp/src/JsModels/paymentFailure.ts
export default class PaymentFailure {
    eventId: number;
    personId?: number | undefined;
    amount: number;
    code: string;
    decline_code?: string | undefined;
    message: string;
    id?: number;
    constructor(
      eventId: number,
      personId: number,
      amount: number,
      code: string,
      decline_code: string,
      message: string,
      id: number
    ) {
      this.eventId = eventId ?? 0;
      this.personId = personId ?? null;
      this.amount = amount ?? "";
      this.code = code ?? "";
      this.decline_code = decline_code ?? "";
      this.message = message ?? "";
      this.id = id ?? - 1;
    }
  }
